gateway-ha/src/main/java/io/trino/gateway/ha/router/HaResourceGroupsManager.java

```diff
@@ -202,32 +184,13 @@ public List<SelectorsDetail> readSelector(long resourceGroupId,
     public SelectorsDetail updateSelector(SelectorsDetail selector, SelectorsDetail updatedSelector,
             @Nullable String routingGroupDatabase)
     {
-        try {
-            connectionManager.open(routingGroupDatabase);
-            String query =
-                    format(
-                            "resource_group_id %s and priority %s "
-                                    + "and user_regex %s and source_regex %s "
-                                    + "and query_type %s and client_tags %s "
-                                    + "and selector_resource_estimate %s",
-                            getMatchingString(selector.getResourceGroupId()),
-                            getMatchingString(selector.getPriority()),
-                            getMatchingString(selector.getUserRegex()),
-                            getMatchingString(selector.getSourceRegex()),
-                            getMatchingString(selector.getQueryType()),
-                            getMatchingString(selector.getClientTags()),
-                            getMatchingString(selector.getSelectorResourceEstimate()));
-            Selectors model = Selectors.findFirst(query);
-
-            if (model == null) {
-                Selectors.create(new Selectors(), updatedSelector);
-            }
-            else {
-                Selectors.update(model, updatedSelector);
-            }
+        SelectorsDao dao = getSelectorsDao(routingGroupDatabase);
+        Selectors model = dao.findFirst(selector);
+        if (model == null) {
+            dao.insert(updatedSelector);
```

Comment:
Seems like a bad idea to silently do an insert instead of an update

Reply:
This logic is same as before. I don't want to change the logic in code refactoring.

Reply:
yeah ... time for another refactor/thinking after this PR maybe.